Islandora/islandora_xml_forms
# ISLANDORA-2205: Re-design `creative_commons` element

Also, implement ISLANDORA-2208 and solve issues in #251

**JIRA Ticket**: https://jira.duraspace.org/browse/ISLANDORA-2205

* https://jira.duraspace.org/browse/ISLANDORA-2208
* #251

# What does this Pull Request do?

Re-design `creative_commons` element to solve bugs and implement new features. In particular, users can set default license in form builder, choose Public Domain licenses, override license versions and work without JS.

# What's new?

* Honour '#default_value'
* Allow CC0 & Public Domain Mark licenses
* Graceful non-JS degradation
* Allow overriding license version
* Allow setting license URI manually
* Validate versions, resp. license URIs
* Cache last query to API for speed-up
* Fix `<html>` wrapper of API response
* Renounce to value callback; populate values in element creation function
* Remove '#tree' logic in favour of manipulating `$form_state` directly

# How should this be tested?

* On a test VM, try to reproduce all the issues listed in the JIRA tickets and the above-mentioned PR.
* Pull this PR
* Verify that issues have disappeared (and no new ones arose).

# Additional Notes:

Since this PR basically re-designs the `creative_commons` element (in particular dropping the '#tree' manipulation), it could well be that
1. This code is more buggy than the previous one
2. There are neater ways of doing what is done here

A thorough code review is, therefore, indispensable.

Example:
* Does this change require documentation to be updated? **perhaps?**
* Does this change add any new dependencies?  **no**
* Does this change require any other modifications to be made to the repository (ie. Regeneration activity, etc.)? **probably not, as functions only got extended by optional arguments**
* Could this change impact execution of existing code? **hopefully not**

# Interested parties
@adam-vessey @bondjimbond @rosiel @whikloj @DiegoPino @Islandora/7-x-1-x-committers
